components/movementsensor/adxl345/adxl345.go

```diff
@@ -232,6 +358,44 @@ func (adxl *adxl345) writeByte(ctx context.Context, register, value byte) error
 	return handle.WriteByteData(ctx, register, value)
 }
 
+func (adxl *adxl345) configureInterruptRegisters(ctx context.Context) error {
+	if adxl.interruptsEnabled == 0 {
+		return nil
+	}
+	adxl.configuredRegisterValues[IntEnableAddr] = adxl.interruptsEnabled
+	adxl.configuredRegisterValues[IntMapAddr] = adxl.interruptsMap
+	for key, value := range defaultRegisterValues {
+		if configuredVal, ok := adxl.configuredRegisterValues[key]; ok {
+			value = configuredVal
+		}
+		err := adxl.writeByte(ctx, key, value)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (adxl *adxl345) readInterrupts(ctx context.Context) {
+	adxl.mu.Lock()
+	defer adxl.mu.Unlock()
+	intSourceRegister, err := adxl.readByte(ctx, IntSourceAddr)
+	if err != nil {
+		adxl.logger.Error(err)
```

Comment:
should return this error and do something?

Reply:
Yeah, I feel like we should have this function return the error, and then in the place where this function is called, we can use `utils.UncheckedError()` or perhaps `utils.UncheckedErrorFunc()` if we cannot return the error further (e.g., because we're in a background goroutine)